data_structures/singly_linked_list2.cpp

```diff
@@ -0,0 +1,313 @@
+/**
+ * @ linked list using template
+ * @brief The linked list stores data in sequential storage, like arrays. Though the data are stored sequentially,
+   the memory locations are not contiguous. Unlike an array, the linked list can store data of different data types.
+
+ * @author Nasreen parween (https://github.com/NasreenParween)
+ * @see singly_linked_list.cpp, stack_using_linked_list.cpp, circular_linked_list.cpp ,doubly_linked_listn.cpp
+ */
+
+#include<iostream>
+using namespace std;
```

Comment:
You've used this but I suggest moving your implementation inside a separate `namespace` under the `data_structures` namespace. For reference, take a look [here](https://github.com/TheAlgorithms/C-Plus-Plus/blob/1371b5390abc98ab127c5a9057cbeff6e2afbbfb/data_structures/linked_list.cpp#L21-L31).